### PR TITLE
3 login

### LINKED ISSUE #3: Log In Page UI

### LINKED ISSUE #4: Sign Up Page UI

### LINKED ISSUE #43: Log In Functionality
Allow users to log in with email and password

### PR DESCRIPTION
**Pull request by MC72144**
closes #3 closes #4 closes #43

### CODE REVIEW

File: app/src/main/java/com/example/feelgoodinc/database/Result.java
Comment (nimitz-Bear) [inline app/src/main/java/com/example/feelgoodinc/database/Result.java:1]: Again, not really sure what this is doing. To me, it looks like generics are being used just to print out the error/success of a class. Wouldn't we be better off just getting the `Result` data directly from the `LoggedInUser` model?

File: app/src/main/java/com/example/feelgoodinc/models/LoggedInUser.java
Comment (nimitz-Bear) [inline app/src/main/java/com/example/feelgoodinc/models/LoggedInUser.java:1]: This file should probably be with the other `app/src/main/Models`.

Also, wouldn't it make more sense to have a generic `User` model, then just have a bool as well for if they're logged in or not?

**Review by nimitz-Bear — CHANGES_REQUESTED** [review]
There are a couple of changes.

- [ ] Also, `LoginActivity` and `SignupActivity` have some backend functions in them. It's best practice to put them into a seperate `Service` called `LoginService` (probably a `Service` bound to an `Activity`)
- [ ] remove any eclipse files/directories like `.settings` to avoid any issues. These should potentially be added to the .gitignore as well to avoid issues in future.
- [ ] Refactor `LoggedInUser`, by:
     - [ ] moving it to the `app/src/main/Models` folder with the other data models
     - [ ] refactoring it to be a general `User` model, which can be re-used better later on (with an `isLoggedIn` bool)
- [ ] Add some comments to the modified `build.gradle` files to delineate where the login/signup firebase dependencies come in
- [ ] (OPTIONAL, but recommended) Create an activity-bound `Service` that combines `LoginDataSource` and `LoginDataRepository`, `Login` Activity's `loginUserAccount()` and Signup's `registerNewUser` into a single `Service`.

File: app/src/main/java/com/example/feelgoodinc/database/JournalDatabaseHelper.java [hunk 1]
```diff
@@ -32,7 +29,7 @@ public class JournalDatabaseHelper {
 
     public JournalDatabaseHelper() {
         firestore = FirebaseFirestore.getInstance();
-        journalsRef = firestore.collection("users").document(User.getCurrentUserKey()).collection("journals");
+        // journalsRef = firestore.collection("users").document(User.getCurrentUserKey()).collection("journals");
```
Comment (nimitz-Bear) [inline app/src/main/java/com/example/feelgoodinc/database/JournalDatabaseHelper.java:32]: I'll probably need to add this back in later on

File: app/src/main/java/com/example/feelgoodinc/services/UserService.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Date;
+
+/***
+ * Accesses the firebase database to receive and add user data
+ * Used for log in, sign up and change password
+ */
+public class UserService extends Service {
```
Comment (nimitz-Bear) [inline app/src/main/java/com/example/feelgoodinc/services/UserService.java:40]: It'd be good to have an example of how to use the service here in the Javadoc. Surround the code example with `<pre>  </pre>`

**Review by nimitz-Bear — APPROVED** [review]
Overall fine, it's just adding more documentation to the `UserService` to make it more clear how this should be used

**Review by nimitz-Bear — APPROVED** [review]

**Review by Woodhouse15 — APPROVED** [review]
👍

**Pull request merged**